pkg/workers/move/export.go

```diff
@@ -0,0 +1,393 @@
+	if err != nil {
+		return
+	}
+	defer func() {
+		if errc := out.Close(); err == nil {
+			err = errc
+		}
+	}()
+
+	gw, err := gzip.NewWriterLevel(out, gzip.BestCompression)
+	if err != nil {
+		return
+	}
+	tw := tar.NewWriter(gw)
+
+	if n, err = writeInstanceDoc(i, "instance", createdAt, tw, nil); err != nil {
+		return
+	}
+	size += n
+
+	settings, err := i.SettingsDocument()
+	if err != nil {
+		return
+	}
+	if n, err = writeDoc("", "settings", settings, createdAt, tw, nil); err != nil {
+		return
+	}
+	size += n
+
+	if !opts.WithoutIndex {
+		var root *vfs.TreeFile
+		root, err = i.VFS().BuildTree()
+		if err != nil {
+			return
+		}
+		n, err = writeDoc("", "files-index", root, createdAt, tw, nil)
+		if err != nil {
+			return
+		}
+		size += n
+
+		cursors, _ := splitFilesIndex(root, nil, BucketSize, BucketSize)
+		exportDoc.IndexFilesCursors = cursors
+	}
+
+	n, err = exportDocs(i, opts.WithDoctypes, createdAt, tw)
+	if errc := tw.Close(); err == nil {
+		err = errc
+	}
+	if errc := gw.Close(); err == nil {
+		err = errc
+	}
+	size += n
+
+	return
+}
+
+// BucketSize is the default size of a file bucket, to split the index into
+// equal-sized parts.
+const BucketSize = 35 * 1024 * 1024 // 50 MB
+
+// splitFilesIndex devides the index into equal size bucket of maximum size
+// `bucketSize`. Files can be splitted into multiple parts to accomodate the
+// bucket size, using a range. It is used to be able to download the files into
+// separate chunks.
+//
+// The method returns a list of cursor into the index tree for each beginning
+// of a new bucket. A cursor has the following format:
+//
+//    ${dirname}/../${filename}-${byterange-start}
+func splitFilesIndex(root *vfs.TreeFile, cursors []string, bucketSize, sizeLeft int64) ([]string, int64) {
+	if root.FilesChildrenSize > sizeLeft {
+		for _, child := range root.FilesChildren {
+			size := child.ByteSize
+			if size <= sizeLeft {
+				sizeLeft -= size
+				continue
+			}
+			size -= sizeLeft
+			for size > 0 {
+				rangeStart := (child.ByteSize - size)
+				cursor := path.Join(root.Fullpath, child.DocName) + ":" + strconv.FormatInt(rangeStart, 10)
+				cursors = append(cursors, cursor)
+				size -= bucketSize
+			}
+			sizeLeft = bucketSize + size
+		}
+	}
+	for _, dir := range root.DirsChildren {
+		cursors, sizeLeft = splitFilesIndex(dir, cursors, bucketSize, sizeLeft)
+	}
+	return cursors, sizeLeft
+}
+
+func exportDocs(in *instance.Instance, withDoctypes []string, now time.Time, tw *tar.Writer) (size int64, err error) {
+	doctypes, err := couchdb.AllDoctypes(in)
+	if err != nil {
+		return
+	}
+	for _, doctype := range doctypes {
+		if len(withDoctypes) > 0 && !utils.IsInArray(doctype, withDoctypes) {
+			continue
+		}
+		switch doctype {
+		case consts.Jobs, consts.KonnectorLogs,
+			consts.Archives,
+			consts.OAuthClients, consts.OAuthAccessCodes:
+			// ignore these doctypes
+		case consts.Sharings, consts.SharingsAnswer:
+			// ignore sharings ? TBD
```

Comment:
Yes, we should ignore sharings for the moment. And we should add `consts.Shared` to this list too.